src/manifest.rs

```diff
@@ -8,11 +8,25 @@ pub(crate) struct Manifest {
 
 impl Manifest {
     pub(crate) fn parse(path: std::path::PathBuf) -> anyhow::Result<Self> {
+        // Normally, we'd like to parse directly via `cargo_toml::Manifest::from_path()`.
+        // Using the path is preferable to parsing from a string, because auto-detection of
+        // surrounding files is sometimes necessary to determine the existence of lib targets
+        // and ensure proper handling of workspace inheritance.
+        //
+        // However, `cargo_toml` currently has buggy handling of renamed library targets:
+        // https://gitlab.com/crates.rs/cargo_toml/-/merge_requests/16/
+        //
+        // This is a workaround for that bug, added in:
+        // https://github.com/obi1kenobi/cargo-semver-checks/pull/371/files#diff-a44cbf177b0e747788be2cc21913a967ce8a76f371b98a6c17cd908c400bf1a9
         let manifest_text = std::fs::read_to_string(&path)
             .map_err(|e| anyhow::format_err!("Failed when reading {}: {}", path.display(), e))?;
-        let parsed = toml::from_str(manifest_text.as_str())
-            .map_err(|e| anyhow::format_err!("Failed to parse {}: {}", path.display(), e))?;
-
+        let parsed = match toml::from_str(&manifest_text) {
+            Ok(parsed) => parsed,
+            Err(_) => {
+                // Parsing failed, possibly because of workspace inheritance. Retry with cargo_toml.
+                cargo_toml::Manifest::from_path(&path)?
+            }
+        };
```

Comment:
Why do two phase loading instead of directly loading what is needed?

Reply:
cargo_toml failed to get the correct library name, and therefore this is a workaround to forcefully overwrite the library name parsed by cargo_toml. I have an upstream PR. Once it's fixed we can remove this workaround.

Reply:
@epage is there something I could add or change about the comment above this block to make this clearer? I updated it [here](https://github.com/obi1kenobi/cargo-semver-checks/pull/371/commits/19e66b291a12c8c3a61f5c1561add0578393c31f) but it sounds like it was still confusing.

Reply:
I had missed it because the comment is not on the weird case but on the expected case.

Reply:
I'll try to move it closer 👍